packages/eas-cli/src/metadata/apple/config/writer.ts

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * Serializes the Apple ASC entities into the metadata configuration schema.
+ * This uses version 0 of the config schema.
+ */
+export class AppleConfigWriter {
+  constructor(public readonly schema: Partial<AppleMetadata> = {}) {}
+
+  /** Get the schema result to write it to the config file */
+  toSchema(): { configVersion: number; apple: Partial<AppleMetadata> } {
+    return {
+      configVersion: 0,
+      apple: this.schema,
+    };
+  }
+
+  setAgeRating(attributes: AttributesOf<AgeRatingDeclaration>): void {
+    this.schema.advisory = attributes;
+  }
+
+  setInfoLocale(attributes: AttributesOf<AppInfoLocalization>): void {
+    this.schema.info = this.schema.info ?? {};
+    const existing = this.schema.info[attributes.locale] ?? {};
+
+    this.schema.info[attributes.locale] = {
+      ...existing,
+      title: attributes.name ?? 'no name provided',
+      subtitle: optional(attributes.subtitle),
+      privacyPolicyUrl: optional(attributes.privacyPolicyUrl),
+      privacyPolicyText: optional(attributes.privacyPolicyText),
+      privacyChoicesUrl: optional(attributes.privacyChoicesUrl),
+    };
+  }
+
+  setCategories({ primaryCategory, secondaryCategory }: AttributesOf<AppInfo>): void {
+    this.schema.categories = [];
+
+    // TODO: see why these types are conflicting
+    if (primaryCategory) {
+      this.schema.categories.push(primaryCategory.id as any);
+      if (secondaryCategory) {
+        this.schema.categories.push(secondaryCategory.id as any);
+      }
+    }
+  }
+
+  setVersion(
+    attributes: Omit<AttributesOf<AppStoreVersion>, 'releaseType' | 'earliestReleaseDate'>
+  ): void {
+    this.schema.copyright = optional(attributes.copyright);
+  }
+
+  setVersionRelease(
+    attributes: Pick<AttributesOf<AppStoreVersion>, 'releaseType' | 'earliestReleaseDate'>
+  ): void {
+    if (attributes.releaseType === ReleaseType.SCHEDULED) {
+      this.schema.release = {
+        autoReleaseDate: optional(attributes.earliestReleaseDate),
+      };
+    }
+
+    if (attributes.releaseType === ReleaseType.AFTER_APPROVAL) {
+      this.schema.release = {
+        automaticRelease: true,
+      };
+    }
+
+    if (attributes.releaseType === ReleaseType.MANUAL) {
+      this.schema.release = {
+        automaticRelease: false,
+      };
+    }
+  }
+
+  setVersionLocale(attributes: AttributesOf<AppStoreVersionLocalization>): void {
+    this.schema.info = this.schema.info ?? {};
+    const existing = this.schema.info[attributes.locale] ?? {};
+
+    this.schema.info[attributes.locale] = {
+      ...existing,
+      description: optional(attributes.description),
+      keywords: optional(attributes.keywords)
+        ?.split(',')
+        .map(keyword => keyword.trim()),
+      releaseNotes: optional(attributes.whatsNew),
+      marketingUrl: optional(attributes.marketingUrl),
+      promoText: optional(attributes.promotionalText),
+      supportUrl: optional(attributes.supportUrl),
```

Comment:
in most cases, it does not matter, but we usually prefer the syntax
`...(attributes.supportUrl ? { supportUrl: attributes.supportUrl } : {})`
in case a key with undefined value is treated differently than when the key is missing. It would be rare for it to work this way, but with above approach you do not need to think about it.

Reply:
Yep, agreed! In this case, we use the data from the writer to serialize everything to JSON. And that is being handled properly:

<img width="261" alt="Screenshot 2022-06-07 at 18 45 19" src="https://user-images.githubusercontent.com/1203991/172437571-79491365-7c26-4e8b-aa5c-9633eb58e66a.png">

Not sure if it's worth making the code harder to read/maintain and adding nested object creations + object spreads for each individual property?

Reply:
Also, one of the reasons for this is that the types in apple utils mostly require `null` or a value. E.g.

<img width="310" alt="Screenshot 2022-06-07 at 18 48 05" src="https://user-images.githubusercontent.com/1203991/172438103-de9798ec-eada-45b9-a72c-5ff92cf708ef.png">

So, if we want to use these properties for models, we have to use `null` instead. And if we want to omit those "non-values" in JSON, we'd have to use `undefined`.